src/components/DropList/DropList.utils.js

```diff
@@ -201,43 +201,36 @@ export function getEnabledItemIndex({
 }) {
   let enabledItemIndex = 0
 
-  if (arrowKey === 'UP') {
-    for (let index = items.length - 1; index >= 0; index--) {
-      const isGroupOrDividerItem = checkIfGroupOrDividerItem(items[index])
-
-      if (
-        !isGroupOrDividerItem &&
-        !items[index].isDisabled &&
-        (nextHighlightedIndex === 0 || index <= nextHighlightedIndex)
-      ) {
-        enabledItemIndex = index
-        break
-      }
-    }
-  }
+  const isNextIndexItemHighlightable =
+    !checkIfGroupOrDividerItem(items[nextHighlightedIndex]) &&
+    !items[nextHighlightedIndex].isDisabled
 
-  if (arrowKey === 'DOWN') {
-    if (currentHighlightedIndex === nextHighlightedIndex) {
-      return getEnabledItemIndex({
-        currentHighlightedIndex,
-        nextHighlightedIndex: 0,
-        items,
-        arrowKey,
-      })
+  if (
+    isNextIndexItemHighlightable &&
+    currentHighlightedIndex !== nextHighlightedIndex
+  ) {
+    enabledItemIndex = nextHighlightedIndex
+  } else {
+    let newNextHighlightedIndex
+
+    if (arrowKey === 'UP') {
+      newNextHighlightedIndex =
+        nextHighlightedIndex - 1 >= 0
```

Comment:
I think this is something like `isNotLast` - could we maybe extract it to variable saying what exactly is checked here? Would be easier to get what is going on here :)
And the same on line 223